muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/guider/GuiderRESTService.java

```diff
@@ -688,6 +697,216 @@ private GuiderStudentWorkspaceActivityRestModel toRestModel(GuiderStudentWorkspa
         assessmentState);
     return model;
   }
+  /**
+   * POST mApi().guider.student.contactLog.create(userEntityId, payload)
+   * 
+   * Creates a new contact log entry from the given payload and studentEntityId.
+   * 
+   * Payload: 
+   * 
+   * {text: "something something",
+   *  entryDate: 2021-02-15
+   *  creatorName: "Etunimi Sukunomi",
+   *  type: "OTHER"/"LETTER"/"EMAIL"/"PHONE"/"CHATLOG"/"SKYPE"/"FACE2FACE"/"ABSENCE"/"MUIKKU"
+   * }
+   *  
+   * Output: Created student contact log entry
+   * 
+   * {
+   * id: 123;
+   * text: "something something",
+   * creatorId: 23,
+   * creatorName: "Etunimi Sukunimi";
+   * entryDate: 2021-02-15;
+   * type: "PHONE";
+   * }
+   */
+  @POST
+  @Path("/student/{ID}/contactLog/")
+  @RESTPermit (handling = Handling.INLINE, requireLoggedIn = true)
```

Comment:
No permission handling within the method at all. Perhaps...

`@RESTPermit (GuiderPermissions.ACCESS_CONTACT_LOG)`

...given to all staff roles. Same with all the other endpoints dealing with contact log entries.